File: insight/insight-maven/src/main/scala/io/fabric8/insight/maven/aether/ConsoleRepositoryListener.scala
/**
 *  Copyright 2005-2014 Red Hat, Inc.
 *
 *  Red Hat licenses this file to you under the Apache License, version
 *  2.0 (the "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 *  implied.  See the License for the specific language governing
 *  permissions and limitations under the License.
 */
package io.fabric8.insight.maven.aether

import org.sonatype.aether.{RepositoryEvent, AbstractRepositoryListener}
import org.slf4j.LoggerFactory

class ConsoleRepositoryListener extends AbstractRepositoryListener {

  val LOG = LoggerFactory.getLogger(classOf[ConsoleRepositoryListener])
  
  override def artifactDeployed(event: RepositoryEvent) =
    LOG.debug("Deployed " + event.getArtifact + " to " + event.getRepository)

  override def artifactDeploying(event: RepositoryEvent) =
    LOG.debug("Deploying " + event.getArtifact + " to " + event.getRepository)

  override def artifactInstalled(event: RepositoryEvent) =
    LOG.debug("Installed " + event.getArtifact + " to " + event.getFile)

  override def artifactInstalling(event: RepositoryEvent) =
    LOG.debug("Installing " + event.getArtifact + " to " + event.getFile)

  override def artifactResolved(event: RepositoryEvent) =
    LOG.debug("Resolved " + event.getArtifact + " from " + event.getRepository)

  override def artifactResolving(event: RepositoryEvent) =
    LOG.debug("Resolving " + event.getArtifact + " from " + event.getRepository)


  override def artifactDescriptorMissing(event: RepositoryEvent) =
    LOG.warn("Missing artifact descriptor for " + event.getArtifact)

  override def artifactDescriptorInvalid(event: RepositoryEvent) =
    LOG.warn("Invalid artifact descriptor for " + event.getArtifact +
            ": " + event.getException().getMessage)


  override def metadataDeployed(event: RepositoryEvent) =
    LOG.debug("Deployed " + event.getMetadata + " to " + event.getRepository)

  override def metadataDeploying(event: RepositoryEvent) =
    LOG.debug("Deploying " + event.getMetadata + " to " + event.getRepository)

  override def metadataInstalled(event: RepositoryEvent) =
    LOG.debug("Installed " + event.getMetadata + " to " + event.getFile)

  override def metadataInstalling(event: RepositoryEvent) =
    LOG.debug("Installing " + event.getMetadata + " to " + event.getFile)

  override def metadataResolved(event: RepositoryEvent) =
    LOG.debug("Resolved " + event.getMetadata + " from " + event.getRepository)

  override def metadataResolving(event: RepositoryEvent) =
    LOG.debug("Resolving " + event.getMetadata + " from " + event.getRepository)


  override def metadataInvalid(event: RepositoryEvent) =
    LOG.warn("Invalid metadata " + event.getMetadata)

}

File: insight/insight-maven/src/main/java/io/fabric8/insight/maven/aether/ConsoleRepositoryListener.java
/**
 *  Copyright 2005-2014 Red Hat, Inc.
 *
 *  Red Hat licenses this file to you under the Apache License, version
 *  2.0 (the "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 *  implied.  See the License for the specific language governing
 *  permissions and limitations under the License.
 */
package io.fabric8.insight.maven.aether;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.eclipse.aether.AbstractRepositoryListener;
import org.eclipse.aether.RepositoryEvent;

public class ConsoleRepositoryListener extends AbstractRepositoryListener {

    private static final Logger LOG = LoggerFactory.getLogger(ConsoleRepositoryListener.class);

    @Override
    public void artifactDeployed(RepositoryEvent event) {
        LOG.debug("Deployed " + event.getArtifact() + " to " + event.getRepository());
    }

    @Override
    public void artifactDeploying(RepositoryEvent event) {
        LOG.debug("Deploying " + event.getArtifact() + " to " + event.getRepository());
    }

    @Override
    public void artifactDescriptorInvalid(RepositoryEvent event) {
        LOG.warn("Invalid artifact descriptor for " + event.getArtifact() +
            ": " + event.getException().getMessage());
    }

    @Override
    public void artifactDescriptorMissing(RepositoryEvent event) {
        LOG.warn("Missing artifact descriptor for " + event.getArtifact());
    }

    @Override
    public void artifactInstalled(RepositoryEvent event) {
        LOG.debug("Installed " + event.getArtifact() + " to " + event.getFile());
    }

    @Override
    public void artifactInstalling(RepositoryEvent event) {
        LOG.debug("Installing " + event.getArtifact() + " to " + event.getFile());
    }

    @Override
    public void artifactResolved(RepositoryEvent event) {
        LOG.debug("Resolved " + event.getArtifact() + " from " + event.getRepository());
    }

    @Override
    public void artifactResolving(RepositoryEvent event) {
        LOG.debug("Resolving " + event.getArtifact() + " from " + event.getRepository());
    }

    @Override
    public void metadataDeployed(RepositoryEvent event) {
        LOG.debug("Deployed " + event.getMetadata() + " to " + event.getRepository());
    }

    @Override
    public void metadataDeploying(RepositoryEvent event) {
        LOG.debug("Deploying " + event.getMetadata() + " to " + event.getRepository());
    }

    @Override
    public void metadataInstalled(RepositoryEvent event) {
        LOG.debug("Installed " + event.getMetadata() + " to " + event.getFile());
    }

    @Override
    public void metadataInstalling(RepositoryEvent event) {
        LOG.debug("Installing " + event.getMetadata() + " to " + event.getFile());
    }

    @Override
    public void metadataInvalid(RepositoryEvent event) {
        LOG.warn("Invalid metadata " + event.getMetadata());
    }

    @Override
    public void metadataResolved(RepositoryEvent event) {
        LOG.debug("Resolved " + event.getMetadata() + " from " + event.getRepository());
    }

    @Override
    public void metadataResolving(RepositoryEvent event) {
        LOG.debug("Resolving " + event.getMetadata() + " from " + event.getRepository());
    }

}
